packages/serverpod_shared/test/config_test.dart

```diff
@@ -0,0 +1,329 @@
+import 'package:serverpod_shared/serverpod_shared.dart';
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
+
+void main() {
+  test(
+      'Given Serverpod config missing api server configuration when loading from Yaml then exception is thrown.',
+      () {
+    expect(
+      () => ServerpodConfig.loadFromYaml('', '', {}, YamlMap()),
+      throwsA(isA<Exception>().having(
+        (e) => e.toString(),
+        'message',
+        equals('Exception: apiServer is missing in config'),
+      )),
+    );
+  });
+
+  test(
+      'Given Serverpod config with api server configuration missing required port when loading from Yaml then exception is thrown.',
+      () {
+    var serverpodConfig = '''
+apiServer:
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+''';
+
+    expect(
+      () => ServerpodConfig.loadFromYaml(
+        '',
+        '',
+        {},
+        loadYaml(serverpodConfig),
+      ),
+      throwsA(isA<Exception>().having(
+        (e) => e.toString(),
+        'message',
+        equals(
+            'Exception: apiServer is missing required configuration for port.'),
+      )),
+    );
+  });
+
+  test(
+      'Given Serverpod config with api server with wrong port type when loading from Yaml then exception is thrown.',
+      () {
+    var serverpodConfig = '''
+apiServer:
+  port: '8080'
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+''';
+
+    expect(
+      () => ServerpodConfig.loadFromYaml(
+        '',
+        '',
+        {},
+        loadYaml(serverpodConfig),
+      ),
+      throwsA(isA<Exception>().having(
+        (e) => e.toString(),
+        'message',
+        equals(
+            'Exception: apiServer configuration has invalid type for port. Expected int, got String.'),
+      )),
+    );
+  });
+
+  test(
+      'Given Serverpod config with api server configuration when loading from Yaml then config is loaded correctly.',
+      () {
+    var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+''';
+
+    var config = ServerpodConfig.loadFromYaml(
+      'myRunMode',
+      'myServerId',
+      {'serviceSecret': 'password'},
+      loadYaml(serverpodConfig),
+    );
+
+    expect(config.runMode, 'myRunMode');
+    expect(config.serverId, 'myServerId');
+    expect(config.apiServer.port, 8080);
+    expect(config.apiServer.publicHost, 'localhost');
+    expect(config.apiServer.publicPort, 8080);
+    expect(config.apiServer.publicScheme, 'http');
+    expect(config.insightsServer, isNull);
+    expect(config.webServer, isNull);
+    expect(config.maxRequestSize, 524288);
+    expect(config.serviceSecret, 'password');
+    expect(config.database, isNull);
+    expect(config.redis, isNull);
```

Comment:
Suggestion: Refactor this test into a group and specific tests for each slice, this makes it easier to add / remove things in the future if the structure changes and/or grows and shrinks. Ideally we shouldn't have to edit expects but rather delete / remove the entire test, leads to less risk of changing the wrong thing.

Reply:
Apply this logic on all tests.